hail/python/hail/nd/nd.py

```diff
@@ -269,6 +259,52 @@ def solve(a, b, no_crash=False):
     return result
 
 
+@typecheck(nd_coef=expr_ndarray(), nd_dep=expr_ndarray(), lower=expr_bool)
+def solve_tri(nd_coef, nd_dep, lower=False):
```

Comment:
Let's give it the same name as scipy: https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.solve_triangular.html

Two reasons: 
1. It's good to be consistent with other tools
2. `tri` could mean `tridiagonal`, which is a different thing.